### PR TITLE
Manual Expense Tracking Feature

### PR DESCRIPTION
**Pull request by jonahansmulcrone**
built out functionality and ui components for a manual expense tracking feature. users can now create income transactions and regular (expense) transactions based on historical revenue/spending.

### CODE REVIEW

**Review by StuartGavidia — CHANGES_REQUESTED** [review]

File: AnalyticsService/app/models.py [hunk 1]
```diff
@@ -3,15 +3,63 @@
...
+    __tablename__ = 'regular_transaction'
+
+    transaction_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    company_id = db.Column(db.String(255), nullable=False)
+    account_name = db.Column(db.String(255), nullable=False)
+    amount = db.Column(db.Integer, nullable=False)
+    descriptions = db.Column(db.String(255), nullable=False)
+    budget_id = db.Column(db.Integer, nullable=False)
+
+class IncomeTransaction(db.Model):
```
[the remark this answers is not in the document]
Comment (jonahansmulcrone) [reply AnalyticsService/app/models.py:54]: there will only be two types of transactions, regular (expenses) and income transactions. It's basically a self-reporting system for expenses and income.

File: AnalyticsService/app/routes.py [hunk 2]
```diff
@@ -24,24 +29,27 @@ def create_budget():
...
     occurance = data.get('occurance', '')
 
     new_budget = Budget(
-        #company_id=company_id,
+        company_id=company_id,
         account_name=account_name,
         allowance=allowance,
         budget_date=budget_date,
-        occurance=occurance
+        occurance=occurance,
```
[the remark this answers is not in the document]
Comment (jonahansmulcrone) [reply AnalyticsService/app/routes.py:51]: thanks for pointing that out!

File: client/src/features/Analytics/Analytics.tsx [hunk 3]
```diff
@@ -116,28 +254,76 @@ const Analytics: React.FC = () => {
...
+                                    <Row>
+                                        <Table style={{
+                                            boxShadow: '0 4px 8px rgba(0, 0, 0, 0.1)',
+                                            padding: '10px',
+                                            borderRadius: '15px',
+                                            justifyContent: 'center',
+                                        }}>
+                                            <thead>
+                                                <tr>
+                                                    <th style={{ fontWeight: 'bold', border: '1px solid #dddddd', textAlign: 'left', padding: '8px' }}>Account Name</th>
```
[the remark this answers is not in the document]
Comment (jonahansmulcrone) [reply client/src/features/Analytics/Analytics.tsx:306]: good call, I'll move the code to a css file

File: client/src/features/Analytics/BudgetCard.tsx [hunk 4]
```diff
@@ -1,7 +1,12 @@
 import { Card, ProgressBar } from "react-bootstrap"
```
[the remark this answers is not in the document]
Comment (jonahansmulcrone) [reply client/src/features/Analytics/BudgetCard.tsx:1]: noted! I'll take care of this.

**Review by StuartGavidia — APPROVED** [review]
Looks good, nice pr!

**Pull request merged**